### PR TITLE
Add mirage support principal managed groups

### PR DESCRIPTION
**Pull request by DhariniJeeva**
This PR adds mocks for managed groups in `role/principals list` and `add to principals` view for both global and org-level scope. 

This is part of https://hashicorp.atlassian.net/browse/ICU-7102 

global scope: 
Screen Shot 2022-12-20 at 1 37 24 PM

org Scope: 
Screen Shot 2022-12-20 at 1 36 50 PM

### CODE REVIEW

File: addons/api/mirage/factories/role.js [hunk 1]
```diff
@@ -28,9 +28,13 @@ export default factory.extend({
   withPrincipals: trait({
     afterCreate(role, server) {
       const { scope } = role;
+      const { id: scopeId } = scope;
       const users = server.createList('user', 2, { scope });
       const groups = server.createList('group', 2, { scope });
-      role.update({ users, groups });
+      const managedGroups = server.schema.managedGroups
+        .where({ scopeId })
+        .models.map((model) => model);
```
Comment (randallmorey) [inline addons/api/mirage/factories/role.js:36]:
```suggestion
        .models;
```

File: addons/api/mirage/scenarios/default.js [hunk 2]
```diff
@@ -51,8 +51,43 @@ export default function (server) {
...
+      (auth) => auth.type === 'oidc'
+    );
+    const managedGroups = server.createList('managed-group', 2, {
+      scope,
+      authMethodId: oidcAuthMethod[0].id,
+    });
+    role.update({ managedGroups });
+  });
+
+  const OrgScopeRoles = server.createList(
```
Comment (randallmorey) [inline addons/api/mirage/scenarios/default.js:73]:
```suggestion
  const orgScopeRoles = server.createList(
```

File: addons/api/mirage/scenarios/default.js [hunk 3]
```diff
@@ -51,8 +51,43 @@ export default function (server) {
...
+    'role',
+    5,
+    { scope: orgScope },
+    'withPrincipals'
+  );
+  //create managed groups for role/principals in orgScope
+  OrgScopeRoles.forEach((role) => {
```
Comment (randallmorey) [inline addons/api/mirage/scenarios/default.js:80]:
```suggestion
  orgScopeRoles.forEach((role) => {
```

File: addons/api/mirage/scenarios/default.js [hunk 4]
```diff
@@ -51,8 +51,43 @@ export default function (server) {
...
+    1,
+    { scope: globalScope },
+    'withPrincipals'
+  );
+  //create managed groups for role/principals in globalScope
+  globalScopeRoles.forEach((role) => {
+    const { scope } = role;
+    const oidcAuthMethod = globalAuthMethods.filter(
+      (auth) => auth.type === 'oidc'
+    );
```
Comment (randallmorey) [inline addons/api/mirage/scenarios/default.js:65]:
```suggestion
    )[0];
```

File: addons/api/mirage/scenarios/default.js [hunk 5]
```diff
@@ -51,8 +51,43 @@ export default function (server) {
...
+    const managedGroups = server.createList('managed-group', 2, {
+      scope,
+      authMethodId: oidcAuthMethod[0].id,
```
Comment (randallmorey) [inline addons/api/mirage/scenarios/default.js:68]:
```suggestion
      authMethodId: oidcAuthMethod.id,
```

**Review by randallmorey — CHANGES_REQUESTED** [review]
Made some minor suggestions.  One question I had is why model managed groups with a `scope`?  IIRC this doesn't appear in the API.

**Comment by DhariniJeeva** [comment]
> Made some minor suggestions. One question I had is why model managed groups with a `scope`? IIRC this doesn't appear in the API.

ah, you are right. we don't need scope id for the managed group, updated!

**Comment by DhariniJeeva** [comment]
> Made some minor suggestions. One question I had is why model managed groups with a `scope`? IIRC this doesn't appear in the API.

actually that's a valid point, I'm double checking if everything works without the scope Id since this was already there in the code

**Comment by DhariniJeeva** [comment]
> > Made some minor suggestions. One question I had is why model managed groups with a `scope`? IIRC this doesn't appear in the API.
> 
> actually that's a valid point, I'm double checking if everything works without the scope Id since this was already there in the code

After an offline discussion, it is understood that we add scopes to managed groups.

**Review by ZedLi — APPROVED** [review]

**Pull request merged**